### DISCUSSION TITLE
Ban specific functions, similar to flake8-tidy-imports (TID)

### DISCUSSION THREAD

**Discussion by billy-doyle** [astral-sh/ruff#9236] [Ideas]
Working on a codebase with some use of `globals()` and `locals()` (with no reason to have those used). Was wondering if there was any way to ban specific function use in ruff, similar to how https://github.com/tripcher/flake8-forbidden-func works.

As an example, tidy imports does the following in the toml:
```
[tool.ruff.lint.flake8-tidy-imports.banned-api]
"importlib".msg = "Banned package import importlib"
```

To ban specific built-ins (or really any function from any package), would propose something like:
```
[tool.ruff.lint.flake8-forbidden-func.banned-api]
"globals()".msg = "Banned function globals"
"locals()".msg = "Banned function locals"
```

**Comment by roberthoenig** [comment]
Any update on this?

**Reply by MichaReiser** [reply]
You can use https://docs.astral.sh/ruff/settings/#lint_flake8-tidy-imports_banned-api but I'm not sure if it supports builtins (like `locals` and `globals`)

**Reply by roberthoenig** [reply]
Thanks for the link - this seems to work only for banning modules though? (So not for functions  in general)

**Reply by MichaReiser** [reply]
It should work for functions that you import from other modules. I don't think it works for builtins.

> Specific modules or module members that may not be imported or accessed.

**Comment by qris** [comment]
It would be really nice to have this, e.g. to prevent `breakpoint()` calls getting into master.

**Reply by billy-doyle** [reply]
https://docs.astral.sh/ruff/rules/debugger/ covers that, but yea additional bans would be nice
